src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,297 @@
+
+  static get styles() {
+    return css`
+      .text {
+        cursor: grab;
+        user-select: none;
+      }
+      input:focus + svg #container {
+        stroke: #4e50d7;
+        stroke-width: 3;
+      }
+      .hide-input {
+        position: absolute;
+        clip: rect(0 0 0 0);
+        width: 1px;
+        height: 1px;
+        margin: -1px;
+      }
+      .dialer-anim {
+        transform-origin: center;
+      }
+      @keyframes dialOne {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(56deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialTwo {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(87deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialThree {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(115deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialFour {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(143deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialFive {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(173deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialSix {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(204deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialSeven {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(232deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialEight {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(260deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+      @keyframes dialNine {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(290deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+
+      @keyframes dialZero {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(320deg);
+        }
+        100% {
+          transform: rotate(0);
+        }
+      }
+    `;
+  }
+
+  private static matchDigit(
+    slots: SVGPathElement,
+    digit: Digit,
+    direction: Direction = Direction.normal,
+    duration = '1000ms',
+    timingFn = 'ease-out'
+  ) {
+    switch (digit) {
+      case 0:
+        slots?.setAttribute('style', `animation: dialZero ${duration} 1 ${direction} ${timingFn};`);
```

Comment:
There is a lot of repetition here, and if I understand correctly, you can replace the switch statement with an object that will map from digit to its name, or rename the animations to dial0, dial1, etc. so you can use the same line for all the animations.

Also, you can probably avoid the direct DOM manipulating by changing some instance variable here, and then using this variable in the `render()` function so set the respective style on the slots element.

Reply:
Thats a very good feedback, if there was a certain math pattern that would be best cause than I was just iterating the numbers and increasing the degrees according to a const or a calc result but since I couldn't find one I did the switch-case.
 
At first when I had the reverse I passed different inputs to the functions but now it's definitely not DRY and you're right I can use an object mapper, I'll think I'll try to switch to transition with classes instead of animation and style - that will change the current implementation, if I'll eventually stay with animation keyframes I'll change that function according to your comment :)